common/ext/class.GenerisInstaller.php

```diff
@@ -46,12 +50,24 @@ public function install()
         }
  
         $this->installLoadDefaultConfig();
+        $this->setupSchema();
         $this->installOntology();
         $this->installRegisterExt();
-        
-        common_cache_FileCache::singleton()->purge();
+
+        $this->getServiceManager()->get(common_cache_Cache::SERVICE_ID)->purge();
         
         $this->log('d', 'Installing custom script for extension ' . $this->extension->getId());
         $this->installCustomScript();
     }
+
+    public function setupSchema()
+    {
+        $pm = $this->getServiceManager()->get(PersistenceManager::SERVICE_ID);
+        $schemaCollection = $pm->getSqlSchemas();
+        $ontology = $this->getServiceManager()->get(Ontology::SERVICE_ID);
+        if ($ontology instanceof SchemaAwareInterface) {
```

Comment:
So when the ontology is not implementing SchemaAwareInterface we are not going to touch schema and we are silent about that?

Reply:
yes, as this indicates that our ontology implementation does not require SQL schemas (maybe we implemented a graph solution)